domain/user/state/state.go

```diff
@@ -0,0 +1,335 @@
+
+	addUserQuery := "INSERT INTO user (uuid, name, display_name, created_by_uuid, created_at) VALUES (?, ?, ?, ?, ?)"
+
+	_, err := tx.ExecContext(ctx, addUserQuery,
+		uuid.String(),
+		user.Name,
+		user.DisplayName,
+		creatorUUID.String(),
+		time.Now(),
+	)
+	if databaseutils.IsErrConstraintUnique(err) {
+		return usererrors.AlreadyExists
+	} else if databaseutils.IsErrConstraintForeignKey(err) {
+		return usererrors.UserCreatorUuidNotFound
+	} else if err != nil {
+		return fmt.Errorf("cannot insert user %q: %w", user.Name, err)
+	}
+
+	return nil
+}
+
+func setPasswordHash(ctx context.Context, tx *sql.Tx, userUUID domainuser.UUID, passwordHash string, salt []byte) error {
+	defineUserAuthenticationQuery := `
+INSERT INTO user_authentication (user_uuid, disabled) 
+VALUES (?, ?) 
+ON CONFLICT(user_uuid) DO UPDATE SET user_uuid = excluded.user_uuid, disabled = excluded.disabled
+`
+
+	_, err := tx.ExecContext(ctx, defineUserAuthenticationQuery, userUUID, false)
+	if err != nil {
+		return fmt.Errorf("cannot set authentification for user %q: %w", userUUID, err)
+	}
+
+	setPasswordHashQuery := `
+INSERT INTO user_password (user_uuid, password_hash, password_salt) 
+VALUES (?, ?, ?) 
+ON CONFLICT(user_uuid) DO UPDATE SET password_hash = excluded.password_hash, password_salt = excluded.password_salt
+`
+
+	_, err = tx.ExecContext(ctx, setPasswordHashQuery, userUUID.String(), passwordHash, salt)
+	if err != nil {
+		return fmt.Errorf("cannot set password hash for user %q: %w", userUUID, err)
+	}
+
+	return nil
+}
+
+func removePasswordHash(ctx context.Context, tx *sql.Tx, userUUID domainuser.UUID) error {
+	removePasswordHashQuery := `
+DELETE FROM user_password
+WHERE user_uuid = ?
+`
+
+	_, err := tx.ExecContext(ctx, removePasswordHashQuery, userUUID.String())
+	if err != nil {
+		return fmt.Errorf("cannot remove password hash for user %q: %w", userUUID, err)
+	}
+
+	return nil
+}
+
+func getUser(ctx context.Context, tx *sql.Tx, userUUID domainuser.UUID) (domainuser.User, error) {
+	getUserQuery := `
+SELECT name, display_name, created_by_uuid, created_at
+FROM user
+WHERE uuid = ? AND removed = false
+`
+
+	var domainUser domainuser.User
+	row := tx.QueryRowContext(ctx, getUserQuery, userUUID)
+
+	var name, displayName string
+	var creatorUUID domainuser.UUID
+	var createdAt time.Time
+	err := row.Scan(&name, &displayName, &creatorUUID, &createdAt)
+	if err != nil {
+		return domainuser.User{}, usererrors.NotFound
+	}
+	domainUser = domainuser.User{
+		Id:          userUUID,
+		Name:        name,
+		DisplayName: displayName,
+		CreatorUUID: creatorUUID,
+		CreatedAt:   createdAt,
+	}
+
+	return domainUser, nil
+}
+
+func setActivationKey(ctx context.Context, tx *sql.Tx, userUUID domainuser.UUID, activationKey string) error {
+	defineUserAuthenticationQuery := `
+INSERT INTO user_authentication (user_uuid, disabled) 
+VALUES (?, ?) 
+ON CONFLICT(user_uuid) DO UPDATE SET user_uuid = excluded.user_uuid, disabled = excluded.disabled
```

Comment:
See comments above about the disabled field being a security problem. On conflict we just want todo nothing.